File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxLen = 0; 
        boolean present[] = new boolean[128];
        int li = 0;
        
        for(int ri=0 ; ri<s.length() ; ri++){
            char curr = s.charAt(ri);
            
            while(present[curr]){
                present[s.charAt(li)] = false;
                li++;
            }
            
            present[curr] = true;
            maxLen = Math.max(maxLen , ri-li+1);            
        }
        
        return maxLen;
        
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> set;
        int li = 0, maxLen = 0;

        for(int ri = 0; ri<s.size() ; ri++){
            while(set.count(s[ri])>0){
                set.erase(s[li]);
                li++;
            }

            set.insert(s[ri]);
            maxLen = max(maxLen, ri-li+1);
        }
        
        return maxLen;
    }
};